package.json

```diff
@@ -172,6 +172,9 @@
     "webpack-cli": "^3.1.0",
     "workbox-build": "^3.6.3"
   },
+  "resolutions": {
+    "**/hoist-non-react-statics": "^3.2.1"
```

Comment:
It's just safer this way. The major bump was never needed in the first place so all this essentially does is backporting all 3.x changes to 2.x.

Reply:
I find the major bump very interesting. It's ensuring people won't have a broken version.
For the resolutions part, I have one concern with the fact it might hide some problem when having multiple version installed.

Reply:
That's what I wanted to address. It forces other packages to also pull in 3.x. There was no actual breaking change between 2.x and 3.x. It "just" patches the unmaintained packages. Another analogy is that the version of hnrs version `2.x || 3.x` is equivalent to the semantic version `^2.0.0`. It would have the same effect on the node_modules layout. The yarn docs describe the use case in more detail.

The issue of multiple versions is not specific to hnrs. If a package uses a broken dependency then it should be fixed. Either by patching upstream or using a patched fork (or other non-conventional methods like patch-package).

I also want to emphasize that this (intentionally) does not affect users. How they configure their build is their concern. This is only for our workspace.

Reply:
I'm all in if it can make our life easier in this challenging React update API periode.